### PR TITLE
Sign Windows binaries with SignPath certificate

### PR DESCRIPTION
**Pull request by janisozaur**
This PR adds automated, cloud-based executable signing with SignPath Foundation's certificate.

SignPath Foundation allows open source projects to use their certificate for signing executables. I have submitted OpenRCT2 as OSS project that would like to use their infrastructure and they agreed to sponsor us, among other projects.

There are some restrictions on how we can use it, they're all explained at https://signpath.org/terms, here's a brief summary:
1. We need to have an established project ownership (i.e. who can push to repository, and the need for review; we already have that: https://github.com/OpenRCT2/OpenRCT2/blob/develop/contributors.md#development-team)
2. The developers need to be using 2FA (GitHub should already that from everyone: https://github.blog/news-insights/product-news/raising-the-bar-for-software-security-github-2fa-begins-march-13/)
3. The builds need to come from trusted CI environment (e.g. GitHub actions; we already have that)
4. The binaries need to have their product name and version attributes set (this PR fixes that)
5. We can only sign our binaries (so no signing of DLLs built as our dependencies; we already adhere)
6. We need to have a privacy policy (this PR adds one)
7. We need to have a code signing policy (this PR adds one)
8. We need to name SignPath as provider of the certificates (this PR adds one)
9. We need to have SignPath listed at https://openrct2.io/ download page **this is not done yet**
10. **We can only sign manually scheduled builds** (this PR adds support for that)

Overall this means we can do almost one-click signed releases now with a certificate that should be more trusted than my own. The last point serves partially as a quota limiter, but more importantly ensures that we are actually aware of the code in repository and are not signing any malware.

An example of fully automated, signed release I created for testing this PR: https://github.com/janisozaur/OpenRCT2/releases/tag/v0.222.4, created in https://github.com/janisozaur/OpenRCT2/actions/runs/11928597564/job/33246249750

To create such a manual run, you need to go to CI workflow (pinned): https://github.com/OpenRCT2/OpenRCT2/actions/workflows/ci.yml and when this PR is merged, there will be a button to schedule a manual run. I have implemented safeguards to drop out of the job if the certificate selected is incorrect or somehow certificate gets requested when not scheduled manually. I have already updated this repository with the necessary API token.

I'm submitting this PR as draft, since the SignPath foundation only enable test certificate initially and then when process is in place they do a review and enable release certificate.

Note: This is **not** OpenRCT2 certificate, this is SignPath certificate they allow OSS projects to use.

### CODE REVIEW

**Comment by janisozaur** [comment]
Once our code review is done, I will invite SignPath to take a look and hopefully releasing 0.4.17 will be a breeze. I can invite team member to SignPath as well.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -587,6 +587,10 @@ jobs:
         uses: actions/download-artifact@v4
         with:
           merge-multiple: true
+      # Having multiple artifacts named the same might be onfusing to the users. Drop the unsigned versions
```
Comment (tupaschoal) [inline .github/workflows/ci.yml:640]: Typo on "onfusing"

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -587,6 +587,10 @@ jobs:
...
+      - name: Remove unsigned artifacts
+        run: |
+          rm -rf *unsigned*
```
Comment (tupaschoal) [inline .github/workflows/ci.yml:643]: Should we add an if to do this only for windows?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -643,11 +643,16 @@ jobs:
...
+          echo "Release created in ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}" >> release_notes.txt
+          echo "" >> release_notes.txt
+
           echo "SHA1 checksums:" >> release_notes.txt
           echo "\`\`\`" >> release_notes.txt
           cat OpenRCT2-${{ needs.build_variables.outputs.name }}-sha1sums.txt >> release_notes.txt
           echo "\`\`\`" >> release_notes.txt
           echo "" >> release_notes.txt
+
+          cat release_notes.txt
```
Comment (tupaschoal) [inline .github/workflows/ci.yml:665]: I'm assuming this is for debug only and will go away

File: readme.md [hunk 4]
```diff
@@ -246,12 +247,22 @@ We would also like to distribute additional scenarios with the game, when the ti
...
 ---
 
-# 5. Licence
+# 5. Code signing policy
+
+We sign our releases digital certificate provided by SignPath Foundation.
+
+Free code signing provided by [SignPath.io](https://about.signpath.io/), certificate by [SignPath Foundation](https://signpath.org/).
+
+Signed releases can only be done by member of the [development team](https://github.com/OpenRCT2/OpenRCT2/blob/develop/contributors.md#development-team).
```
Comment (tupaschoal) [inline readme.md:265]: by members

File: readme.md [hunk 5]
```diff
@@ -246,12 +247,22 @@ We would also like to distribute additional scenarios with the game, when the ti
 
 ---
 
-# 5. Licence
+# 5. Code signing policy
+
+We sign our releases digital certificate provided by SignPath Foundation.
```
Comment (tupaschoal) [inline readme.md]: This reads weird, should it be something like:
```
We sign our releases with a digital certificate provided by SignPath Foundation.
```

**Review by tupaschoal — COMMENTED** [review]
This is a great improvement. I looked through each commit and it seems fine (though I did abstract away some stuff)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (janisozaur) [reply .github/workflows/ci.yml:643]: No need in my opinion. All the artifacts are collected in this job and only matches would be the ones to be removed

File: .github/workflows/ci.yml (hunk 3 above)
Comment (janisozaur) [reply .github/workflows/ci.yml:665]: I'd like it to stay. It makes the job much easier in case something goes wrong at a later stage

**Comment by GingerAdonis** [comment]
Unfortunately no cert for dev builds. This is great nonetheless!

**Comment by janisozaur** [comment]
I'd like this not to be squashed

File: PRIVACY.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+- No tracking or analytics services are implemented
+
+## Crash Reporting
+- Crash dumps may be sent to Backtrace.io for diagnostic purposes
+- Each crash report transfer requires explicit user consent
+- Crash dumps contain:
+  - System information
+  - Machine hostname
+  - System username
+  - Technical data about the crash
```
Comment (duncanspumpkin) [inline PRIVACY.md:30]: Should we mention in game screenshot

File: scripts/build [hunk 7]
```diff
@@ -12,25 +12,23 @@ cd $basedir
...
+    echo "#define OPENRCT2_FILE_VERSION $fileversion" > "resources/version.h"
+    echo "#define OPENRCT2_PRODUCT_VERSION \"$productversion\"" >> "resources/version.h"
+    cat "resources/version.h"
 
     # Build everything
     echo -e "\033[0;36mBuilding OpenRCT2 for Windows $CONFIGURATION|$PLATFORM...\033[0m"
     vstool msbuild openrct2.proj -t:build -p:Breakpad=true -p:UsePCH=true
 
     # Create openrct2.exe and openrct2.com with correct subsystem
     cp bin/openrct2.exe bin/openrct2.com
```
Comment (IntelOrca) [inline scripts/build:30]: `openrct2.exe` is copied to `openrct2.com`, but then not modified to be a console subsystem.
I see you have moved the openrct2.com creation to the signing process, but doesn't this mean dev builds no longer have a correct version of openrct2.com?

File: PRIVACY.md (hunk 6 above)
Comment (janisozaur) [reply PRIVACY.md:30]: Added

File: scripts/build (hunk 7 above)
Comment (janisozaur) [reply scripts/build]: There's no need for the commit that modified it. I left it when I was debugging the signing process, but it works even on files named `.com`

**Comment by janisozaur** [comment]
Addressed comments from @IntelOrca and @duncanspumpkin , here's a release done with the new changes: https://github.com/janisozaur/OpenRCT2/releases/tag/v0.222.5

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Gymnasiast) [reply .github/workflows/ci.yml:640]: Not yet addressed.

```suggestion
      # Having multiple artifacts named the same might be confusing to the users. Drop the unsigned versions.
```

File: distribution/windows/install.nsi [hunk 8]
```diff
@@ -204,6 +207,7 @@ Section "!OpenRCT2" Section1
     CreateShortCut "$SMPROGRAMS\$SHORTCUTS\Readme.lnk" "$INSTDIR\Readme.txt"
     CreateShortCut "$SMPROGRAMS\$SHORTCUTS\Changelog.lnk" "$INSTDIR\Changelog.txt"
     CreateShortCut "$SMPROGRAMS\$SHORTCUTS\Contributors.lnk" "$INSTDIR\contributors.md"
+    CreateShortCut "$SMPROGRAMS\$SHORTCUTS\PrivacyPolicy.lnk" "$INSTDIR\PRIVACY.md"
```
Comment (Gymnasiast) [inline distribution/windows/install.nsi:210]:
```suggestion
    CreateShortCut "$SMPROGRAMS\$SHORTCUTS\Privacy Policy.lnk" "$INSTDIR\PRIVACY.md"
```

File: distribution/windows/install.nsi [hunk 9]
```diff
@@ -240,12 +244,14 @@ Section "Uninstall"
     Delete "$SMPROGRAMS\$SHORTCUTS\Readme.lnk"
     Delete "$SMPROGRAMS\$SHORTCUTS\Changelog.lnk"
     Delete "$SMPROGRAMS\$SHORTCUTS\Contributors.lnk"
+    Delete "$SMPROGRAMS\$SHORTCUTS\PrivacyPolicy.lnk"
```
Comment (Gymnasiast) [inline distribution/windows/install.nsi:247]:
```suggestion
    Delete "$SMPROGRAMS\$SHORTCUTS\Privacy Policy.lnk"
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (janisozaur) [reply .github/workflows/ci.yml:640]: Odd, I did fix it in the web version of vscode (https://GitHub.dev), but it must've been lost somehow when commiting.

**Review by Gymnasiast — COMMENTED** [review]
Seems all right to me.

**Comment by janisozaur** [comment]
This is ready for review, I will ask SignPath to take a look

**Review by tupaschoal — APPROVED** [review]

**Review by ZehMatt — APPROVED** [review]

**Comment by janisozaur** [comment]
This got all the necessary approvals from our team and was approved by SignPath as well. 

I've filed an issue for @Gymnasiast to update our download page: https://github.com/OpenRCT2/OpenRCT2/issues/23280

**Pull request merged**